Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.cpp

```diff
@@ -300,8 +300,8 @@ OptionSet<WebEventModifier> modifiersForNavigationAction(const NavigationAction&
 Page* WebChromeClient::createWindow(LocalFrame& frame, const WindowFeatures& windowFeatures, const NavigationAction& navigationAction)
 {
 #if ENABLE(FULLSCREEN_API)
-    if (frame.document() && frame.document()->fullscreenManager().currentFullscreenElement())
```

Comment:
You can check against `fullscreenElement()` which queries the top layer, so it should also take in account the case where the parent iframe is being fullscreened.

`currentFullscreenElement()` is a bit flaky because it's only set for explicit requestFullscreen requests.

Reply:
Are you suggesting that we fix this by just updating the check rather than calling `cancelFullscreen` unconditionally? After this patch, there is currently no check at all in this file.

`cancelFullscreen` already checks `fullscreenElement`.

Though, I realized calling `topDocument()` here is unnecessary, since `cancelFullscreen` does that too.

Let me know what you think.

Reply:
Looking at `cancelFullscreen`, the current patch is fine. I was trying to avoid an unnecessary call to cancelFullscreen, but it sounds like adding the `fullscreenElement()` check would just be more work.

and yes, invoking `topDocument()` here is unnecessary.